### PR TITLE
[Bug Fix]: form submission not working for radio, checkbox

### PR DESCRIPTION
**Pull request by neerajadobe93**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--<repo>--<owner>.hlx.page/
- After: https://<branch>--<repo>--<owner>.hlx.page/

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -3,10 +3,12 @@ import { sampleRUM } from '../../scripts/lib-franklin.js';
...
-    if (fe.type === 'checkbox') {
-      if (fe.checked) payload[fe.id] = fe.value;
-    } else if (fe.id) {
-      payload[fe.id] = fe.value;
+    if ( fe.type === "radio") {
+        if (fe.checked) payload[fe.name] = fe.value;
+    } else if (fe.type === "checkbox") {
+      if (fe.checked) payload[fe.name] = payload[fe.name] ? `${payload[fe.name]},${fe.value}`: fe.value
+    } else if (fe.name) {
+      payload[fe.name] = fe.value;
```
Comment (vdua) [inline blocks/form/form.js:11]: the check for fe.name should be the first check

```
if (fe.name) {
   if (fe.type === 'radio' && fe.checked) {
         payload[fe.name] = fe.value
   } else if (fe.type === 'checkox' && fe.checked) { .... }
   else {
       ...
   }
}
```

**Review by vdua — CHANGES_REQUESTED** [review]
Please fix lint issues and look at my comment.

File: blocks/form/form.js (hunk 1 above)
Comment (neerajadobe93) [reply blocks/form/form.js:11]: @vdua  i think name field is required for every input field, so ideally it should fail if name is undefined ? please correct if my understanding is correct

Comment (neerajadobe93) [reply blocks/form/form.js:11]: lint issues are fixed with form.js file. varun , i am seeing multiple lint issues as well with other blocks. i am using  airbnb-base-eslint config.

Comment (vdua) [reply blocks/form/form.js:11]: > i think name field is required for every input field
No that is not correct. Consider buttons, output, etc. elements. 

I was just saying that check for `fe.name` that is there in [Line 10](https://github.com/adobe/afb/pull/8/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R10) should be moved up.

**Review by vdua — APPROVED** [review]
LGTM and as discussed please update the if check for fe.name and merge.

**Comment by neerajadobe93** [comment]
@vdua  can you please merge this PR as i don't have write access

**Pull request closed without merging**